src/main/java/com/klodnicki/bike/rest/API/Bike/rest/controller/UserController.java

```diff
@@ -1,12 +1,47 @@
 package com.klodnicki.bike.rest.API.Bike.rest.controller;
 
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RestController;
+import com.klodnicki.bike.rest.API.Bike.exception.NotFoundInDatabaseException;
+import com.klodnicki.bike.rest.API.Bike.model.entity.User;
+import com.klodnicki.bike.rest.API.Bike.service.UserService;
+import org.springframework.web.bind.annotation.*;
 
 @RestController
-@RequestMapping("/user")
+@RequestMapping("/api/user")
 public class UserController {
 
+    private final UserService userService;
+
+    public UserController(UserService userService) {
+        this.userService = userService;
+    }
+
+    @PostMapping("/add")
+    public User addUser(@RequestBody User user) {
+        return userService.addUser(user);
+    }
+
+    @GetMapping("/{id}") //dostawałem 400 bad request bo była zła adnotacja, była @RequestParam a powinna być @PathVariable
+    // i wtedy daję: localhost:8888/api/user/2
+    public User findUserById (@PathVariable("id") Long id) throws NotFoundInDatabaseException {
+        return userService.findUserById(id);
+    }
+
+    @GetMapping
+    public Iterable<User> findAllUsers() {
```

Comment:
Why did you decide to return Iterable of User? What's the reason behind it?

Reply:
Because this is a default method of CrudRepository and for me it was okay to have a return type Iterable
![image](https://github.com/BartoliniAndBorderCollies/Rent_bike_REST_API_Official/assets/126821059/085ae571-2b04-4a49-be52-a30565809396)

Reply:
Sure, so the reason is simple :) Fine by me.